### ISSUE TITLE
Allow scrap by folder name for dos games

### ISSUE THREAD

**Issue by navyseai** [muldjord/skyscraper#270]
I'm not sure if I'm doing this the right approach on this or if I'm doing something wrong but when I try to scrap my ms dos games using TheGamesDB is a mess. I believe this is due the approach of scrapping dos games as a single file rom. 
Using the current process I scrap all kinds of files (ega.exe, dos4gw.exe, demo.bat, so on...)

Some issues I expect to help solve with this approach:
- no longer attempt to scrap bat files or exe files that are complements to the game
- ignore the main files names which sometimes happen to be identical (lets imagine game 1 and game 2 both have d2.exe) causing the identification to be wrong
- prevent doing so many pointless requests to webservice (making the process lot faster)
- Having the data available from the main menu ms-dos page and not the subfolders (like other emulators)

I think my proposal is simple to implement, to use the folder names instead

I also would like to hear your opinions

**Comment by muldjord** [comment]
Yes, the DOS platform is a mess to scrape for a few different reasons:
- Simply scraping the exe files results in several *helper* files also getting scraped (dos4gw.exe is a good example)
- Scraping folders only by name makes it unusable with hashing based sources (ScreenScraper)
- Some emulators relies on script files for running the games (to mount CD images and set up the env), in which case it is the script that should be scraped and *not* either the exe files *or* the folder

So that's what it comes down to. A lot of ugly hardcoding would be necessary for this to work properly. Skyscraper is largely emulator- and platform-agnostic when scraping. I try to avoid a lot of *if this platform then that* or *if this filetype then that* or *if this scraper then quit if file is folder* and that's just not possible to avoid for DOS.

Half a year ago I tried implementing what you are suggesting, and I ended up backtracking most of it due to how ugly and messy it became.

I'm not currently actively developing new features for Skyscraper due to lack of time. But I do fix bugs. I think this lies somewhere between those. So it's up in the air if I get the time and / or motivation to work on this.

Thank you for your suggestions. I'll leave this open.

**Comment by muldjord** [comment]
A quick note by the way. I suggest taking a look at the [extensions](https://github.com/muldjord/skyscraper/blob/master/docs/CONFIGINI.md#extensionszip-uae-adf) config.ini option. That allows you to at least set the extensions that the platform will be using. Just add a section to config.ini like this:
```
[dos]
extensions="*.bat"
```
